src/dynamodb-mcp-server/awslabs/dynamodb_mcp_server/dynamodb_server.py

```diff
@@ -0,0 +1,1096 @@
+    try:
+        client = get_dynamodb_client(region_name)
+        params = {}
+        if max_results:
+            params['MaxResults'] = max_results
+        if next_token:
+            params['NextToken'] = next_token
+        if table_arn:
+            params['TableArn'] = table_arn
+
+        response = client.list_exports(**params)
+        return {
+            'ExportSummaries': response.get('ExportSummaries', []),
+            'NextToken': response.get('NextToken'),
+        }
+    except Exception as e:
+        return {'error': str(e)}
+
+
+@app.tool()
+async def describe_continuous_backups(
+    table_name: str = table_name,
+    region_name: str = Field(default=None, description='The aws region to run the tool'),
+) -> dict:
+    """Returns continuous backup and point in time recovery status for a table."""
+    try:
+        client = get_dynamodb_client(region_name)
+        response = client.describe_continuous_backups(TableName=table_name)
+        return response['ContinuousBackupsDescription']
+    except Exception as e:
+        return {'error': str(e)}
+
+
+@app.tool()
+async def untag_resource(
+    resource_arn: str = resource_arn,
+    tag_keys: List[str] = Field(description='List of tags to remove.', min_length=1),
+    region_name: str = Field(default=None, description='The aws region to run the tool'),
+) -> dict:
+    """Removes tags from a DynamoDB resource."""
+    try:
+        client = get_dynamodb_client(region_name)
+        response = client.untag_resource(ResourceArn=resource_arn, TagKeys=tag_keys)
+        return response
+    except Exception as e:
+        return {'error': str(e)}
+
+
+@app.tool()
+async def tag_resource(
+    resource_arn: str = resource_arn,
+    tags: List[Tag] = Field(description='Tags to be assigned.'),
+    region_name: str = Field(default=None, description='The aws region to run the tool'),
+) -> dict:
+    """Adds tags to a DynamoDB resource."""
+    try:
+        client = get_dynamodb_client(region_name)
+        response = client.tag_resource(ResourceArn=resource_arn, Tags=tags)
+        return response
+    except Exception as e:
+        return {'error': str(e)}
+
+
+@app.tool()
+async def list_tags_of_resource(
+    resource_arn: str = resource_arn,
+    next_token: str = Field(
+        default=None, description='The NextToken from the previous paginated call'
+    ),
+    region_name: str = Field(default=None, description='The aws region to run the tool'),
+) -> dict:
+    """Returns tags for a DynamoDB resource."""
+    try:
+        client = get_dynamodb_client(region_name)
+        params = {'ResourceArn': resource_arn}
+        if next_token:
+            params['NextToken'] = next_token
+
+        response = client.list_tags_of_resource(**params)
+        return {'Tags': response.get('Tags', []), 'NextToken': response.get('NextToken')}
+    except Exception as e:
+        return {'error': str(e)}
+
+
+@app.tool()
+async def delete_table(
```

Comment:
How might one protect tables from deletion?

Reply:
Customer can enable table deletion protection feature using the MCP Server: https://aws.amazon.com/about-aws/whats-new/2023/03/amazon-dynamodb-table-deletion-protection/

Note that this is a generic risk that exist in any existing toolings - operators can accidentally delete a tool in Console/using AWS CLI, etc.

I think there should be best practices for using IAM role to make that easier.